sky/backends/backend_utils.py

```diff
@@ -2647,6 +2808,51 @@ def _refresh_cluster(cluster_name):
     return kept_records
 
 
+# Internal only:
+def download_and_stream_latest_job_log(
```

Comment:
To avoid the circular import I've done some code movement. The main movement is the `include_reserved` argument in the `backend_utils.get_clusters` function, which originally will call the `Controllers.get_group` to identify the reserved clusters. Now I remove this argument and make it always return all clusters; at the same time, I added a new function in `controller_utils` to call `backend_utils.get_clusters` and filter the result.